src/coreclr/scripts/superpmi.py

```diff
@@ -664,19 +664,31 @@ def __init__(self, coreclr_args):
 
         self.coreclr_args = coreclr_args
 
+        self.temp_location = None
+
         # Pathname for a temporary .MCL file used for noticing SuperPMI replay failures against base MCH.
         self.base_fail_mcl_file = None
 
         # The base .MCH file path
         self.base_mch_file = None
 
         # Final .MCH file path
-        self.final_mch_file = None
+        if self.coreclr_args.output_mch_path is not None:
+            self.final_mch_file = os.path.abspath(self.coreclr_args.output_mch_path)
+            final_mch_dir = os.path.dirname(self.final_mch_file)
+            if not os.path.isdir(final_mch_dir):
+                os.makedirs(final_mch_dir)
+        else:
+            # Default directory is the current working directory (before we've changed the directory using "TempDir")
+            default_mch_location = os.path.abspath(os.getcwd())
+            if not os.path.isdir(default_mch_location):
```

Comment:
In what case `os.getcwd()` would return a location that is not a directory? When it is ran from `C:\` or something?

Reply:
Yeah, getcwd should already return an existing directory.